<launch>
    <!-- Laser Scanner -->
    <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode">
        <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
        <param name="serial_baudrate"     type="int"    value="115200"/>
        <param name="frame_id"            type="string" value="laser"/>
        <param name="inverted"            type="bool"   value="false"/>
        <param name="angle_compensate" type="bool" value="true"/>
    </node>

    <!-- TF -->
    <node pkg="robot8_pose_broadcaster" type="pose_broadcaster" name="pose_broadcaster"/>
    <node pkg="robot8_laser_broadcaster" type="laser_broadcaster" name="laser_broadcaster"/>
    <node pkg="robot8_camera_broadcaster" type="camera_broadcaster" name="camera_broadcaster"/>
    <node pkg="robot8_uarm_broadcaster" type="uarm_broadcaster" name="uarm_broadcaster"/>
    <node pkg="robot8_pose_broadcaster" type="loc_broadcaster" name="loc_broadcaster"/>
    
     <!--Keyboard node-->
    <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="keyboard_controller" output="screen" />

    <!-- Motors -->
    <node name="right" pkg="phidgets" type="motor" >
        <param name="serial" value="473021" />
        <param name="name" type="string" value="right"/>
        <param name="frequency" type="int" value="10"/>
    </node>

    <node name="left" pkg="phidgets" type="motor">
        <param name="serial" value="475406" />
        <param name="frequency" type="int" value="10"/>
        <param name="name" type="string" value="left"/>
    </node>

    <!-- Motor controller -->
    <node pkg="robotgroup8_motor_controller" type="motor_controller" name="motor_controller">
        <param name="alpha" type="double" value="1"/>
        <param name="I" type="double" value="0.4"/>
    </node>

    <!-- Map visualization -->
    <arg name="map_file" default="lab_maze_2016.txt" />
    <node pkg="ras_maze_map" type="ras_maze_map_node" name="ras_maze_map_node"  >
         <param name="map_file" value="$(find ras_maze_map)/maps/lab_maze_2016_2.txt"/>
    </node>
    <!-- TF map to world -->
    <node pkg="tf" type="static_transform_publisher" name="odom_to_map"
        args="0.0 0.0 0.0 0 0 0 /world /map 100" />

    <!-- Odometry -->
    <node pkg="ras_odometry" type="ras_odometry_node" name="odometry">
        <param name="x" type="double" value="2.24"/>
        <param name="y" type="double" value="0.2"/>
        <param name="angle" type="double" value="1.5707963"/>
    </node>

    <!-- Localization -->
    <node name="localization" pkg="ras_localization" type="localization" >
        <param name="map_frame" type="string" value="world"/>
        <param name="map_file" type="string" value="/home/ras28/catkin_ws/src/ras_maze/ras_maze_map/maps/lab_maze_2016_2.txt"/>
        <param name="wall_thickness" type="double" value="0.018"/>
        <param name="motion_variance_v" type="double" value="0.000003"/>
        <param name="motion_variance_w" type="double" value="0.0"/>
        <param name="pose_variance_xy" type="double" value="0.04"/>
        <param name="pose_variance_z" type="double" value="0.5"/>
        <param name="laser_variance" type="double" value="0.025"/>
        <param name="particle_quantity" type="int" value="1000"/>
        <param name="weight_combo_method" type="int" value="2"/>
        <param name="x_init" type="double" value="2.24"/>
        <param name="y_init" type="double" value="0.20"/>
        <param name="z_init" type="double" value="1.5707963"/>
        <param name="gaussian_premultiplier" type="double" value="1"/>
        <param name="laser_min_range" type="double" value="0.05"/>
        <param name="tracking" type="bool" value="false"/>
        <param name="interlaced" type="bool" value="true"/>
        <param name="addnoise" type="bool" value="true"/>
        <param name="noiseincallback" type="bool" value="true"/>
        <param name="resample" type="bool" value="true"/>
        <param name="laser_skip" type="int" value="10"/>
    </node>
    
</launch>
